derp/derp_server.go

```diff
@@ -610,26 +614,26 @@ func (s *Server) registerClient(c *sclient) {
 
 	cs.activeClient.Store(c)
 
-	if _, ok := s.clientsMesh[c.key]; !ok {
-		s.clientsMesh[c.key] = nil // just for varz of total users in cluster
+	if _, ok := s.clientsMesh[c.handle]; !ok {
+		s.clientsMesh[c.handle] = nil // just for varz of total users in cluster
 	}
-	s.keyOfAddr[c.remoteIPPort] = c.key
+	s.keyOfAddr[c.remoteIPPort] = c.handle
 	s.curClients.Add(1)
 	if c.isNotIdealConn {
 		s.curClientsNotIdeal.Add(1)
 	}
-	s.broadcastPeerStateChangeLocked(c.key, c.remoteIPPort, c.presentFlags(), true)
+	s.broadcastPeerStateChangeLocked(c.handle, c.remoteIPPort, c.presentFlags(), true)
 }
 
 // broadcastPeerStateChangeLocked enqueues a message to all watchers
 // (other DERP nodes in the region, or trusted clients) that peer's
 // presence changed.
 //
 // s.mu must be held.
-func (s *Server) broadcastPeerStateChangeLocked(peer key.NodePublic, ipPort netip.AddrPort, flags PeerPresentFlags, present bool) {
+func (s *Server) broadcastPeerStateChangeLocked(peer NodeHandle, ipPort netip.AddrPort, flags PeerPresentFlags, present bool) {
 	for w := range s.watchers {
 		w.peerStateChange = append(w.peerStateChange, peerConnState{
-			peer:    peer,
+			peer:    peer.Value(),
```

Comment:
When I got here I stopped to think about if the shallow copy is problematic if it's referenced after GC has already cleaned up the unique map holding the NodeHandle. We may want to add docs to `type NodeHandle` talking about usage of `.Value()` in this context.